llama-index-integrations/embeddings/llama-index-embeddings-solar/llama_index/embeddings/solar/base.py

```diff
@@ -0,0 +1,97 @@
+"""Solar embeddings file."""
+
+import json
+import requests
+from typing import Any, List, Optional
+
+from llama_index.core.base.embeddings.base import (
+    DEFAULT_EMBED_BATCH_SIZE,
+    BaseEmbedding,
+)
+from llama_index.core.bridge.pydantic import Field
+from llama_index.core.callbacks.base import CallbackManager
+from llama_index.core.base.llms.generic_utils import get_from_param_or_env
+
+DEFAULT_SOLAR_API_BASE = "https://api.upstage.ai/v1/solar/embeddings"
+DEFAULT_SOLAR_MODEL = "solar-1-mini-embedding-query"
+
+
+class SolarEmbedding(BaseEmbedding):
+    """Solar embeddings."""
+
+    api_key: str = Field(description="The Solar API key.")
+    api_base: str = Field(
+        default=DEFAULT_SOLAR_API_BASE,
+        description="The base URL for Solar embedding API.",
+    )
+
+    def __init__(
+        self,
+        model_name: str = DEFAULT_SOLAR_MODEL,
+        api_key: str = "",
+        api_base: str = DEFAULT_SOLAR_API_BASE,
+        embed_batch_size: int = DEFAULT_EMBED_BATCH_SIZE,
+        callback_manager: Optional[CallbackManager] = None,
+        **kwargs: Any,
+    ):
+        api_key = get_from_param_or_env("api_key", api_key, "SOLAR_API_KEY", "")
+        super().__init__(
+            model_name=model_name,
+            api_key=api_key,
+            api_base=api_base,
+            embed_batch_size=embed_batch_size,
+            callback_manager=callback_manager,
+            **kwargs,
+        )
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "SolarEmbedding"
+
+    def _get_embedding(self, text: str) -> List[float]:
+        payload = json.dumps({"input": text, "model": self.model_name})
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": "Bearer " + self.api_key,
+        }
+
+        response = requests.request(
+            "POST", self.api_base, headers=headers, data=payload
+        )
```

Comment:
Strongly suggest using the OpenAI Client for the following reasons:
1. It offers numerous beneficial features, including retry, timeout settings etc.
2. The Solar Embedding API is compatible with the error handling provided by the OpenAI Client.